Library/Homebrew/extend/os/linux/keg_relocate.rb

```diff
@@ -50,6 +50,14 @@ def change_rpath(file, old_prefix, new_prefix)
     file.patch!(interpreter: updated[:interpreter], rpath: updated[:rpath])
   end
 
+  def relocate_build_prefix(old_cellar, new_prefix, new_cellar)
+    elf_files.each do |file|
```

Comment:
You can probably implement this generically using `Keg#binary_executable_or_library_files`.

Reply:
That's a great idea, and would let me simplify this considerably by not having to put anything into `extend/os/linux`/`extend/os/mac`.